src/OlFlatStyleParser.ts

```diff
@@ -0,0 +1,297 @@
+
+  flatStyleToGeoStylerFillSymbolizer(flatStyle: FlatStyle): FillSymbolizer {
+    // NOTE: If fillColor is an expression, we cannot detect the opacity
+    const [fillColor, fillOpacity] = OlFlatStyleUtil.isExpression(flatStyle['fill-color'])
+      ? [OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['fill-color']) as Expression<string>]
+      : OlFlatStyleUtil.getColorAndOpacity(flatStyle['fill-color']);
+
+    // TODO add other fill properties
+    return {
+      kind: 'Fill',
+      color: fillColor,
+      opacity: fillOpacity
+    };
+  }
+
+  flatStyleToGeoStylerLineSymbolizer(flatStyle: FlatStyle): LineSymbolizer {
+    const [strokeColor, strokeOpacity] = OlFlatStyleUtil.isExpression(flatStyle['stroke-color'])
+      ? [OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-color']) as Expression<string>]
+      : OlFlatStyleUtil.getColorAndOpacity(flatStyle['stroke-color']);
+
+    // TODO add other stroke properties
+    return {
+      kind: 'Line',
+      color: strokeColor,
+      opacity: strokeOpacity,
+      width: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-width']) as Expression<number>,
+      cap: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-line-cap']) as Expression<CapType>,
+      join: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-line-join']) as Expression<JoinType>,
+      dasharray: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-line-dash']) as Expression<number>[],
+      dashOffset:
+        OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-line-dash-offset']) as Expression<number>,
+      miterLimit: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-miter-limit']) as Expression<number>,
+      perpendicularOffset: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['stroke-offset']) as Expression<number>
+    };
+  }
+
+  flatStyleToGeoStylerTextSymbolizer(flatStyle: FlatStyle): TextSymbolizer {
+    const [textFillColor, textFillOpacity] = OlFlatStyleUtil.isExpression(flatStyle['text-fill-color'])
+      ? [OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-fill-color']) as Expression<string>]
+      : OlFlatStyleUtil.getColorAndOpacity(flatStyle['text-fill-color']);
+
+    const [textStrokeColor, textStrokeOpacity] = OlFlatStyleUtil.isExpression(flatStyle['text-stroke-color'])
+      ? [OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-stroke-color']) as Expression<string>]
+      : OlFlatStyleUtil.getColorAndOpacity(flatStyle['text-stroke-color']);
+
+    // NOTE: If font is an expression, we cannot detect the size
+    const [font, fontSize] = OlFlatStyleUtil.isExpression(flatStyle['text-font'])
+      ? [[OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-font']) as Expression<string>], undefined]
+      : flatStyle['text-font'] ? [
+        [OlStyleUtil.getFontNameFromOlFont(flatStyle['text-font'])],
+        OlStyleUtil.getSizeFromOlFont(flatStyle['text-font'])
+      ] : [];
```

Comment:
This is quite complex and would be more readable with an if statement.

Reply:
done